DigitalLearningSolutions.Web.Tests/Controllers/Register/ClaimAccountControllerTests.cs

```diff
@@ -240,48 +410,109 @@ public void CompleteRegistrationPost_with_password_not_set_returns_NotFound()
             string email,
             string code
         )
         {
+            // Given
+            var model = GivenClaimAccountViewModel(email: email, registrationConfirmationHash: code);
+
             // When
-            var result = controller.CompleteRegistrationPost(email, code);
+            var result = await controller.CompleteRegistration(
+                new ConfirmPasswordViewModel(),
+                model.Email,
+                model.RegistrationConfirmationHash
+            );
 
             // Then
+            ACallToConvertTemporaryUserToConfirmedUserMustNotHaveHappened();
+
             result.Should().BeRedirectToActionResult().WithControllerName("LearningSolutions")
                 .WithActionName("AccessDenied");
         }
 
         [Test]
-        public void CompleteRegistrationPost_with_no_existing_user_redirects_to_AccessDenied()
+        public async Task CompleteRegistrationPost_with_no_existing_user_redirects_to_AccessDenied()
         {
             // Given
+            var model = GivenClaimAccountViewModel();
+
             A.CallTo(
                 () => userDataService.GetUserIdAndCentreForCentreEmailRegistrationConfirmationHashPair(
-                    DefaultEmail,
-                    DefaultCode
+                    model.Email,
+                    model.RegistrationConfirmationHash
                 )
             ).Returns((null, null, null));
 
             // When
-            var result = controller.CompleteRegistrationPost(DefaultEmail, DefaultCode);
+            var result = await controller.CompleteRegistration(
+                new ConfirmPasswordViewModel(),
+                model.Email,
+                model.RegistrationConfirmationHash
+            );
 
             // Then
+            ACallToConvertTemporaryUserToConfirmedUserMustNotHaveHappened();
+
             result.Should().BeRedirectToActionResult().WithControllerName("LearningSolutions")
                 .WithActionName("AccessDenied");
         }
 
         [Test]
-        public void CompleteRegistrationPost_with_logged_in_user_redirects_to_LinkDlsAccount()
+        public async Task CompleteRegistrationPost_with_logged_in_user_redirects_to_LinkDlsAccount()
         {
+            // Given
+            var model = GivenClaimAccountViewModel();
+
             // When
-            var result = controllerWithLoggedInUser.CompleteRegistrationPost(DefaultEmail, DefaultCode);
+            var result = await controllerWithLoggedInUser.CompleteRegistration(
+                new ConfirmPasswordViewModel(),
+                model.Email,
+                model.RegistrationConfirmationHash
+            );
 
             // Then
+            ACallToConvertTemporaryUserToConfirmedUserMustNotHaveHappened();
+
             result.Should().BeRedirectToActionResult().WithActionName("LinkDlsAccount");
         }
 
+        [Test]
+        public async Task CompleteRegistrationPost_with_invalid_model_display_view_with_errors()

```

Comment:
Can we say "with_validation_errors" instead? Currently the name reads as if the user will be redirected to an error page.

Reply:
Ok